RecoTracker/LST/plugins/alpaka/LSTInputProducer.cc

```diff
@@ -0,0 +1,225 @@
+      auto const& seedTracks = iEvent.get(seedToken);
+
+      if (seedTracks.empty())
+        continue;
+
+      // Get seed track refs
+      edm::RefToBaseVector<reco::Track> seedTrackRefs;
+      for (edm::View<reco::Track>::size_type i = 0; i < seedTracks.size(); ++i) {
+        seedTrackRefs.push_back(seedTracks.refAt(i));
+      }
+
+      edm::ProductID id = seedTracks[0].seedRef().id();
+
+      for (size_t iSeed = 0; iSeed < seedTrackRefs.size(); ++iSeed) {
+        auto const& seedTrackRef = seedTrackRefs[iSeed];
+        auto const& seedTrack = *seedTrackRef;
+        auto const& seedRef = seedTrack.seedRef();
+        auto const& seed = *seedRef;
+
+        if (seedRef.id() != id)
+          throw cms::Exception("LogicError")
+              << "All tracks in 'TracksFromSeeds' collection should point to seeds in the same collection. Now the "
+                 "element 0 had ProductID "
+              << id << " while the element " << seedTrackRef.key() << " had " << seedTrackRef.id() << ".";
+
+        const bool seedFitOk = !trackFromSeedFitFailed(seedTrack);
+
+        const TrackingRecHit* lastRecHit = &*(seed.recHits().end() - 1);
+        TrajectoryStateOnSurface tsos =
+            trajectoryStateTransform::transientState(seed.startingState(), lastRecHit->surface(), &mf);
+        auto const& stateGlobal = tsos.globalParameters();
+
+        std::vector<int> hitIdx;
+        for (auto const& hit : seed.recHits()) {
+          int subid = hit.geographicalId().subdetId();
+          if (subid == (int)PixelSubdetector::PixelBarrel || subid == (int)PixelSubdetector::PixelEndcap) {
+            const BaseTrackerRecHit* bhit = dynamic_cast<const BaseTrackerRecHit*>(&hit);
+            const auto& clusterRef = bhit->firstClusterRef();
+            const auto clusterKey = clusterRef.cluster_pixel().key();
+            hitIdx.push_back(clusterKey);
+          } else {
+            throw cms::Exception("LSTInputProducer") << "Not pixel hits found!";
+          }
+        }
+
+        // Fill output
+        see_px.push_back(seedFitOk ? seedTrack.px() : 0);
+        see_py.push_back(seedFitOk ? seedTrack.py() : 0);
+        see_pz.push_back(seedFitOk ? seedTrack.pz() : 0);
+        see_dxy.push_back(seedFitOk ? seedTrack.dxy(bs.position()) : 0);
+        see_dz.push_back(seedFitOk ? seedTrack.dz(bs.position()) : 0);
+        see_ptErr.push_back(seedFitOk ? seedTrack.ptError() : 0);
+        see_etaErr.push_back(seedFitOk ? seedTrack.etaError() : 0);
+        see_stateTrajGlbX.push_back(stateGlobal.position().x());
+        see_stateTrajGlbY.push_back(stateGlobal.position().y());
+        see_stateTrajGlbZ.push_back(stateGlobal.position().z());
+        see_stateTrajGlbPx.push_back(stateGlobal.momentum().x());
+        see_stateTrajGlbPy.push_back(stateGlobal.momentum().y());
+        see_stateTrajGlbPz.push_back(stateGlobal.momentum().z());
+        see_q.push_back(seedTrack.charge());
+        see_hitIdx.push_back(hitIdx);
```

Comment:
```suggestion
        see_hitIdx.push_back(std::move(hitIdx));
```
or
```suggestion
        see_hitIdx.emplace_back(std::move(hitIdx));
```
might avoid a copy of `hitIdx`.